apps/frontend/src/hooks/user/useUsersDataByFilter.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+
+import { BasicUserDetailsFragment, InstrumentFragment } from 'generated/sdk';
+import { useDataApi } from 'hooks/common/useDataApi';
+
+export function useUsersDataByFilter(instrument: InstrumentFragment | null) {
+  const api = useDataApi();
+  const [surName, setSurName] = useState<string>();
+  const [usersData, setUsersData] = useState<{
+    totalCount: number;
+    users: Array<BasicUserDetailsFragment>;
+  }>({
+    totalCount: 0,
+    users: [],
+  });
+
+  useEffect(() => {
+    if (instrument) {
+      api()
+        .getUser({ userId: instrument.managerUserId })
+        .then((data) => {
+          setSurName(data.user?.lastname);
+          api()
+            .getUsers({ filter: surName })
+            .then((data) => {
+              setUsersData(data.users || { totalCount: 0, users: [] });
+            });
+        });
+    }
+  }, [api, surName, instrument]);
```

Comment:
I'm a little confused with the reason beheind to inject surName as dependency to retrigger useEffect hook
Maybe`({ filter: data.user?.lastname}) ` and remove surName stats should be enough?

Reply:
Sorry, for the confusion. Initially added `surName` variable for some reason but later didn't use it, however forgot to remove from here.